### PR TITLE
Unify CVC4_CHECK/CVC4_DCHECK/AlwaysAssert/Assert.

### PR DESCRIPTION
**Pull request by mpreiner**
This commit deprecates the use of the macros `Assert` and always `AlwaysAssert` in favor or `CVC4_DCHECK` and `CVC4_CHECK`.

### CODE REVIEW

File: src/theory/bv/theory_bv_rewrite_rules.h [hunk 1]
```diff
@@ -389,6 +389,7 @@ class RewriteRule {
   /** Actually apply the rewrite rule */
   static inline Node apply(TNode node) {
     Unreachable();
+    abort();
```
Comment (4tXJ7f) [inline src/theory/bv/theory_bv_rewrite_rules.h]:
```suggestion
    SuppressWrongNoReturnWarning;
```

File: src/theory/quantifiers/query_generator.cpp [hunk 2]
```diff
@@ -177,7 +177,7 @@ void QueryGenerator::checkQuery(Node qy, unsigned spIndex)
         ss << "  " << d_vars[i] << " -> " << pt[i] << std::endl;
       }
       ss << "but CVC4 answered unsat!" << std::endl;
-      AlwaysAssert(false, ss.str().c_str());
+      AlwaysAssert(false) << ss.str().c_str();
```
[the remark this answers is not in the document]
Comment (mpreiner) [reply src/theory/quantifiers/query_generator.cpp]: Doesn't work without `.str()`.

File: src/expr/metakind_template.cpp [hunk 3]
```diff
@@ -57,8 +57,7 @@ size_t NodeValueCompare::constHash(const ::CVC4::expr::NodeValue* nv) {
 
   switch(nv->d_kind) {
 ${metakind_constHashes}
-  default:
-    Unhandled(::CVC4::expr::NodeValue::dKindToKind(nv->d_kind));
+default: Unhandled() << ::CVC4::expr::NodeValue::dKindToKind(nv->d_kind);
```
[the remark this answers is not in the document]
Comment (aniemetz) [reply src/expr/metakind_template.cpp]: Had the same issue with clang-format in this template previously...

[the remark this answers is not in the document]
Comment (mpreiner) [reply src/expr/metakind_template.cpp]: Yeah it's a bit weird, if you clang-format the whole block it's fine... but then it formats the place holders with additional spaces: `$ { metakind_constHashes }`

**Pull request merged**